### ISSUE TITLE
create localizations using entityserivce api not working like expected

### ISSUE THREAD

**Issue by BasharOsman** [strapi/strapi#15389]
### Required System information

- Node.js version: v16.16.0
- Yarn version: 1.22.19
- Strapi version: v4.5.5
- Database: SQLite
- Operating system: ubuntu 20

### Describe the bug

create a localizations using entityService api link in one way (TR->EN but not EN ->TR) 

### Steps to reproduce the behavior

1. Create new quickstart project
2. create new Content Type with field name and enable localization for CT and the Field
3. add new locale from settings lets be Turkish (TR) and set it to be default 
4. in src/index.js in bootstrap function create En version then create TR version and add En id in localizations array
5. go check the created data in content manager you will find that the first data in content type is have two locale TR/EN but if you change table locale to EN the first data will have just EN locale

### Expected behavior

the both created data have same locale TR/EN

#### Code
src/index.js
![Screenshot from 2023-01-10 11-32-57](https://user-images.githubusercontent.com/62757809/211500781-0f241f3a-3b03-4d68-91f8-dff417bb515f.png)

#### created TR

![Screenshot from 2023-01-10 11-17-47](https://user-images.githubusercontent.com/62757809/211501143-8ee2d2f1-89eb-40c3-b9d2-a74fad3ba3f5.png)

#### created En

![Screenshot from 2023-01-10 11-17-59](https://user-images.githubusercontent.com/62757809/211501474-daf84b77-6031-4c76-8f2b-4b9654199555.png)

### Code snippets

```
async bootstrap({ strapi }) {
    const newEnTest = await strapi.entityService.create(
      "api::test-ct.test-ct",
      {
        data: {
          name: "test",
          locale: "en",
        },
      }
    );
    const newTrTest = await strapi.entityService.create(
      "api::test-ct.test-ct",
      {
        data: {
          name: "ornek",
          locale: "tr",
          localizations: [newEnTest.id],
        },
      }
    );
  },
```
### Additional context

even if you go to edit View for TR you can change to En locale normally but return from En to Tr by change locale select will redirect to create new TR for EN

### Additional note 

you can see node_modules/@strapi/plugin-i18n/server/services/core-api.js

**Comment by kobeumut** [comment]
# Required System information
Node.js version: v18.10.0
Yarn version: 1.22.19
Strapi version: v4.5.5
Database: SQLite
Operating system: ubuntu 20

# Describe the bug
When creating one more locale with entityService, every locale in "prefix"_localization_links in the db is incompletely paired and the order is always saved as 1. 

# Code snippets
```
const res= await strapi.entityService.create("api::general-translation.general-translation",{
                    data: {
                        localizations:[contentID],
                        key: json.key,
                        value: translated.text,
                        locale: 'he'
                    },
                    populate:{localizations:true}
                })
```

# Additional context
<img width="747" alt="image" src="https://user-images.githubusercontent.com/5182289/211544598-5d2c9782-4424-4d27-9e08-b95606430bd9.png">
<img width="786" alt="image" src="https://user-images.githubusercontent.com/5182289/211544717-441c595a-c7bd-485f-8919-1d1cc3cbd333.png">

**Comment by robec1982** [comment]
# Bug report
## System information
Node.js version: v16.14.2
Yarn version: 1.22.19
Strapi version: v4.11.5
Database: Postgres
Operating system: Windows 10, Alpine, Amazon Linux 2
## Describe the bug
I got the same issue described by @kobeumut and @BasharOsman after ingesting data in Strapi using the REST API. This behavior is not found when creating the localized content in the dashboard.
## Steps to reproduce the behavior
1. Create a new Strapi v4 project using postgres.
2. Add the Spanish (es) locale in settings/localization.
3. Create a new collection with a text field (e.g. "name") and enable localization for this collection and for the field.
5. POST an entry using the API for the created collection with `"locale":"es"`, with the data below.
```
{"data":{
    "publishedAt":null,
    "name":"Nombre en Español",
    "locale":"es"
}}
```
Save the id of the created entry to be linked to the other you will create in the step 6.
6. POST another entry using the API for the created collection with `"locale":"en"`.
```
{"data":{
    "publishedAt":null,
    "name":"Name in English",
    "locale":"en", 
    "localizations":[id_obtained_in_the_previous_post]
}}
```
7. You will find that the data created for the English version is linked to the Spanish one, but it doesn't work in reverse.

**Comment by BasharOsman** [comment]
after is searched in i18n plugin i found this two functions:
```javascript
const {syncLocalizations,syncNonLocalizedAttributes }= strapi.plugin('i18n').service('localizations');
```
and i create function that help me creating localized entry:
```javascript
async createLocalizedEntry({ UID, params }) {
    let localizations = params.data.localizations;
  //check if localizations field is available and every localizations have id 
  //if not create the localized entry
    if (
      localizations?.length &&
      !localizations.map((v) => v?.id).every(Boolean)
    ) {
      const createLocalizations = localizations?.map(async (v) => {
        if (v?.id) return v;
        return strapi.entityService.create(UID, {
          data: v,
          populate: { localizations: {} },
        });
      });
      const res = await Promise.all(createLocalizations);

      localizations = res.map((v) => v?.id);
    }
    const data = {
      ..._.omit(params.data, ['localizations']),
      localizations,
    };

    const entry = await strapi.entityService.create(UID, {
      ...params,
      data,
      populate: { ...params.populate, localizations: {} },
    });
    const i18nServices = strapi.plugin('i18n').service('localizations');

    //link localizations back and forth
    i18nServices.syncLocalizations(entry, { model: strapi.getModel(UID) });

    // update  Non Localized Attributes in all localizations
    i18nServices.syncNonLocalizedAttributes(entry, {
      model: strapi.getModel(UID),
    });
    return entry;
  }
```

example: 
```javascript
createLocalizedEntry({
  UID: "api::test-ct.test-ct",
  params: {
    //files: if there any files too,
    data: {
      title: "test",
      locale: "en",
      localizations: [
        {
          //id: 1, if the id is available then will link it if not will create then link all localizations
          title:"örnek",
          locale: "tr",
        },
      ],
    },
  },
});
```

note: the syncNonLocalizedAttributes fucntion will update the fields that is NonLocalized for all localizations

**Comment by robec1982** [comment]
Thank you @BasharOsman,

I also solved the problem using external calls to the REST API. The issue in my case was that apparently the relationships are not created by passing the secondary translation through its id, as I was trying to do in
```
{"data":{
    "publishedAt":null,
    "name":"Name in English",
    "locale":"en", 
    "localizations":[id_obtained_in_the_previous_post]
}}
```
The effective way is using another, available endpoint for this kind of relationship, as below.
```
curl --location 'https://{{your.domain.com}}/api/{{localized_collection}}/{{id_obtained_for_the_original_entry}}/localizations' \
--header 'Content-Type: application/json' \
--header 'Authorization: Bearer Token created in Strapi' \
--data '{"publishedAt":null,"name":"nombre en Español","locale":"es"}'
```
Note that the endpoint includes the id of the entry in English (default language in my case) and `/localizations` in the end to indicate that you are adding localized entry to the default one.
After doing this I got the relationship in both directions in the Admin dashboard, thus allowing me to switch effectively from English to Spanish and in reverse.

**Comment by BasharOsman** [comment]
yes with end point there no problem but the main issue is to create localizied entry with Entity Service API
